x-pack/plugins/observability/public/hooks/slo/use_clone_slo.ts

```diff
@@ -33,34 +32,35 @@ export function useCloneSlo() {
     },
     {
       onMutate: async ({ slo, originalSloId }) => {
-        // Cancel any outgoing refetches (so they don't overwrite our optimistic update)
-        await queryClient.cancelQueries(sloKeys.lists());
+        await queryClient.cancelQueries({ queryKey: sloKeys.lists(), exact: false });
 
-        const latestQueriesData = (
-          queryClient.getQueriesData<FindSLOResponse>(sloKeys.lists()) ?? []
-        ).at(0);
-        const [queryKey, data] = latestQueriesData ?? [];
+        const queriesData = queryClient.getQueriesData<FindSLOResponse>({
+          queryKey: sloKeys.lists(),
+          exact: false,
+        });
```

Comment:
getQueriesData returns an array of tuple [queryKey, queryData]. We get the first one which is the latest query.